src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -412,31 +409,90 @@ def output_schema(self) -> Type[BaseModel]:
         """
         return TextGenerationOutput
 
+    def _process_generation_config(
+        self, generation_config: [None, str, pathlib.Path, Dict, GenerationConfig]
+    ) -> GenerationConfig:
+        """
+        Process and return a GenerationConfig. The function can take in a filepath
+        pointing to a json consisting of the config values, a dictionary with the config
+        values, or a loaded GenerationConfig object. If None is given, the defaults are,
+        the pipeline GenerationConfig is used, if provided. If both are None, default
+        are used for generation.
+
+        :param generation_config: either a json filepath, dictionary or loaded
+        GenerationConfig object
+
+        :return: GenerationConfig object or None
+
+        """
+        if isinstance(generation_config, GenerationConfig):
+            return generation_config
+
+        if not generation_config:
+            return None
+
+        # TODO: move to tmp folder
+        if isinstance(generation_config, dict):
+            config_dir = os.getcwd()
+            config_name = "generation_config.json"
+            local_config_path = os.path.join(config_dir, config_name)
+            _LOGGER.info(
+                "Dictionary provided for the generation config. Creating temporary "
+                " generation_config.json"
+            )
+            with open(local_config_path, "w") as f:
+                json.dump(generation_config, f)
```

Comment:
is there no way to load from in-memory dict?

Reply:
the GenerationConfig api requires a path/string to load from. Could potentially move to a tmpfs?